src/imperative/imperative_utils.h

```diff
@@ -982,10 +994,11 @@ inline void CreateEngineOpSeg(
 
     // Create opr segment for previous nodes.
     if (stop && nid > seg_start) {
+      opr_name = idx[seg_start].source->attrs.name;
```

Comment:
I think you can move this and the other statements below a bit more into the condition to avoid the operation in case the condition is not met.